Push image to ghcr

gassmoeller opened the pull request:
#4039 added the TACC docker image, and also included a try to upload the image to ghcr.io instead of docker hub. While the login was successful, the job didnt actually label the image for upload to ghcr.io. This PR represents an extension of that PR in that it:
- actually builds and pushes the TACC docker image to ghcr.io, and
- also uses ghcr.io for the regular docker image

It still also builds the images for docker hub, so existing workflows work as before. I would like to test if the new workflow runs correctly, before discussing if we want to drop Docker hub, or just keep using both.

tjhei commented on .github/workflows/docker.yml:
```diff
@@ -44,7 +51,9 @@ jobs:
-          tags: geodynamics/aspect:latest
+          tags: |
+            ${{ github.repository_owner }}/${{ github.repository }}:latest
+            ghcr.io/${{ github.repository_owner }}/${{ github.repository }}:latest
```
I am not sure if this matters, but does that mean that forks of aspect now automatically upload to the individual's ghcr?
What happens if dockerhub fails, will it still push to ghcr? Should ghcr maybe come first (more likely to be reliable than dockerhub)?

gassmoeller commented:
I have addressed your comments.

>    I am not sure if this matters, but does that mean that forks of aspect now automatically upload to the individual's ghcr?

No, because all jobs are guarded by `if: github.repository == 'geodynamics/aspect'`. However, I have still replaced the name of the organization with `geodynamics`, mostly because there was some issue with the permission of the GITHUB_TOKEN to access that variable.

If the testers are happy, this should be ready.
